/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   main.cpp                                           :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: asarandi <asarandi@student.42.us.org>      +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2018/01/15 17:12:52 by asarandi          #+#    #+#             */
/*   Updated: 2018/01/16 01:42:34 by asarandi         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <exception>
#include "Bureaucrat.hpp"
#include "Form.hpp"
#include "ShrubberyCreationForm.hpp"
#include "RobotomyRequestForm.hpp"
#include "PresidentialPardonForm.hpp"
#include "Intern.hpp"

int	main()
{


	std::cout << "---------------------------------------------------" << std::endl;

	try
	{
		Intern someRandomIntern;
		Form* rrf;
		rrf = someRandomIntern.makeForm("robotomy request", "Bender");
	}
	catch (std::exception & e)
	{
		std::cout << "caught exception!! " << e.what() << std::endl << std::endl;
	}



	try
	{
		Intern someRandomIntern;
		Form* rrf;
		rrf = someRandomIntern.makeForm("roajsdhlaksmy request", "Fender");
	}
	catch (std::exception & e)
	{
		std::cout << "caught exception!! " << e.what() << std::endl << std::endl;
	}


	try
	{
		Intern someRandomIntern;
		Form* rrf;
		rrf = someRandomIntern.makeForm("shrubbery creation", "Bender");
	}
	catch (std::exception & e)
	{
		std::cout << "caught exception!! " << e.what() << std::endl << std::endl;
	}



	try
	{
		Intern someRandomIntern;
		Form* rrf;
		rrf = someRandomIntern.makeForm("creation shrubbery", "Fender");
	}
	catch (std::exception & e)
	{
		std::cout << "caught exception!! " << e.what() << std::endl << std::endl;
	}



	try
	{
		Intern someRandomIntern;
		Form* rrf;
		rrf = someRandomIntern.makeForm("presidential pardon", "Bender");
	}
	catch (std::exception & e)
	{
		std::cout << "caught exception!! " << e.what() << std::endl << std::endl;
	}



	try
	{
		Intern someRandomIntern;
		Form* rrf;
		rrf = someRandomIntern.makeForm("presidential blah bla", "Fender");
	}
	catch (std::exception & e)
	{
		std::cout << "caught exception!! " << e.what() << std::endl << std::endl;
	}








	return 0;
}
